other/EXAMPLES.md

```diff
@@ -1,3 +1,62 @@
 # Examples
 
-There aren't any examples yet! Want to add one? See `CONTRIBUTING.md`
+## Usage with React/Next.js
+
+### Use node's os module to get the user that ran the build and display their username.
+
+```jsx
+import React from 'react'
+import preval from 'preval.macro'
+
+const whoami = preval`
+  const userInfo = require('os').userInfo()
+  module.exports = userInfo.username
+`
+
+export default WhoAmI
+
+function WhoAmI () {
+  return (
+    <div style={{display: 'flex', justifyContent: 'center'}}>
+      <h1>
+        <pre>
+          whoami: {whoami}
+        </pre>
+      </h1>
+    </div>
+  )
+}
+```
+
+### Code reuse with babel-plugin-preval (Especially useful with [Next.js](https://github.com/zeit/next.js))
+
+A small `getPosts.js` node module that loads in markdown files in a "posts" directory.
+```js
+const frontMatter = require('yaml-front-matter');
+const fs = require('fs');
+const path = require('path');
+
+const posts = fs
+  .readdirSync('./posts/')
+  .filter(file => path.extname(file) === '.md')
+  .map(file => frontMatter.loadFront(`./posts/${file}`, 'body'));
+
+module.exports = posts || [];
+```
+
+Preval the `getPosts.js` module so that posts loaded and exported.
+```js
+import preval from 'preval.macro';
+
+const getPostsCode = preval`
+  const fs = require('fs');
+  const code = fs.readFileSync('./getPosts.js', 'utf8');
+  module.exports = code.toString();
+`;
+
+const posts = preval`${getPostsCode}`;
```

Comment:
Actually, I think you should be able to do:

```js
const posts = preval`module.exports = require('./getPosts)`
```

That should simplify this considerably 😀